### PR TITLE
Add a streaming serial protocol sketch.

### PR DESCRIPTION
**Pull request by bsutherland**
Hi Maarten,

Thanks for the amazing design!! I bought a board based on your design on eBay.

I've added a simple serial protocol to allow playback via the serial port. This allows music to be played back from a PC without the need for an SD card via the USB port, which I guess is a bit more convenient.

I've also added a Python script which demonstrates the protocol by playing back IMF files.

Cheers.

### CODE REVIEW

File: examples/SerialIface/play_imf.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Stream an IMF file via the serial interface of the Arduino OPL shield.
```
Comment (bsutherland) [inline examples/SerialIface/play_imf.py]: Good idea! I'll add usage in the case that no parameters are provided.

File: examples/SerialIface/play_imf.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  # Type-2 format
+  if song_length == 0:
+    song_length = 65535
+    f.read(2)
+
+  i = 0
+  while i < song_length:
+    cmd = f.read(4)
+
+    addr, data, delay_cycles = struct.unpack_from('BBH', cmd)
```
Comment (bsutherland) [inline examples/SerialIface/play_imf.py]: Thanks, I only tried Wolf3d and Commander Keen. I'll try to figure it out.

**Comment by bsutherland** [comment]
@DhrBaksteen Thanks for the review, that should take care of it.

File: examples/SerialIface/SerialIface.ino [hunk 3]
```diff
@@ -1,17 +1,17 @@
 /**
  * This is an example sketch from the OPL2 library for Arduino.
  * It streams commands to the OPL2 via the serial port.
- *
```
Comment (bsutherland) [inline examples/SerialIface/SerialIface.ino]: Just deleting trailing whitespace in this file.

**Comment by DhrBaksteen** [comment]
Thanks for the update. I'll merge it

**Comment by bsutherland** [comment]
Thanks!
I didn't know you were planning on selling them yourself. Look forward to seeing it.
Just my opinion, but if you haven't finalised the design, I'd personally prefer a 3.5 mm socket to a terminal block for the output.

**Comment by DhrBaksteen** [comment]
I should receive the second prototype within a few days where I replaced the screw terminals with a 3.5mm jack. Still messing around with the little amplifier a bit to make it sound good :)

**Comment by bsutherland** [comment]
Looking forward to seeing it!
I'm waiting to get my hands on something that accepts line-level input so I can record on my laptop (it only has a mic input :(

**Pull request merged**